test/types/connection.test-d.ts

```diff
@@ -0,0 +1,25 @@
+import { expectError, expectType } from 'tsd';
+
+import { type Connection, type Document, MongoDBResponse, ns } from '../mongodb';
+
+declare const connection: Connection;
+
+expectType<Document>(await connection.command(ns('a'), { cmd: 1 }));
+expectType<Document>(await connection.command(ns('a'), { cmd: 1 }, undefined));
+expectType<Document>(await connection.command(ns('a'), { cmd: 1 }, { socketTimeoutMS: 1 }));
+// TODO fix TS or simplify arguments
```

Comment:
todo for this PR or future ticket?

Reply:
Either, probably this PR, I'll reach out in the team channel, I am having trouble getting the API to properly return `Document` vs the 4th argument's type. The issue is for now, we have all our usages calling this function with 3 arguments so we get back document but our layers of the driver are going to need to ferry the correct type based on that third argument. CryptoConnection.command is an example. So I need a way to type something that passes in a 4th argument that is potentially nullish accurately (at some point, not an blocker for this PR).

Reply:
Cleaned up the overloads, we don't need to support this usage.